src/rust/protocol/redis/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+mod request;
+mod response;
+mod result;
```

Comment:
Storage feels like it can easily be moved out. The result stuff potentially becomes a little awkward when separated from the protocol - we'll be moving stats like `get_key_hit` and `set_not_stored` out of the protocol crate, since we don't know the meaning of the response unless it's paired with the request in the 'result' type. 

Let's try and see how it feels

Reply:
In a fully dynamic metric system (where metrics are added on demand) this is usually solved by namespace as well, e.g. (`set/not_stored`). Since we are using static metrics, it needs to be a little different. But assuming we can modify the metric macros to allow namespace (that is statically expanded at compile time), we might be able to declare the metrics within the protocol crate but increment them elsewhere? For `get` we can create a few derived metrics for each response type, `get_key_hit`, `get_key_miss`, `get_ex` or (incompatible with previous versions but more obviously namespaced) `get/key_hit`, `get/key_miss`, `get/ex`. Since whatever service module we move the logic to will be dependent on the protocol crate, we can update them there.

Would this work?

Reply:
I believe that will work. I need to think about the macro side of it a bit more. The metrics identifier is going to be something like `PARSE_GET`. Since the metrics all have to be public statics anyway, it should be fine to define ones for request-response pairs (eg, get_key_hit) in the protocol crate and update them elsewhere.

Reply:
Oh right, forgot about that. Man would be nice if dynamic scoping of metrics is free :P